extension/app/inpage/ts/inpage.ts

```diff
@@ -331,25 +332,48 @@ function injectEthereumIntoWindow() {
 	}
 
 	const on = async (kind: OnMessage, callback: any) => {
-		console.log(`set on: ${kind}`)
 		switch (kind) {
 			case 'accountsChanged':
-				window.interceptor.onAccountsChangedCallBack = callback as (accounts: string[]) => void
+				window.interceptor.onAccountsChangedCallBacks.add( callback as (accounts: string[]) => void )
 				return
 			case 'message':
-				window.interceptor.onMessageCallBack = callback as (message: ProviderMessage) => void
+				window.interceptor.onMessageCallBacks.add(callback as (message: ProviderMessage) => void)
 				return
 			case 'connect':
-				window.interceptor.onConnectCallBack = callback as (connectInfo: ProviderConnectInfo) => void
+				window.interceptor.onConnectCallBacks.add(callback as (connectInfo: ProviderConnectInfo) => void)
 				return
 			case 'close': //close is deprecated on eip-1193 by disconnect but its still used by dapps (MyEtherWallet)
-				window.interceptor.onDisconnectCallBack = callback as (error: ProviderRpcError) => void
+				window.interceptor.onDisconnectCallBacks.add(callback as (error: ProviderRpcError) => void)
 				return
 			case 'disconnect':
-				window.interceptor.onDisconnectCallBack = callback as (error: ProviderRpcError) => void
+				window.interceptor.onDisconnectCallBacks.add(callback as (error: ProviderRpcError) => void)
 				return
 			case 'chainChanged':
-				window.interceptor.onChainChangedCallBack = callback as (chainId: string) => void
+				window.interceptor.onChainChangedCallBacks.add(callback as (chainId: string) => void)
+				return
+			default:
+		}
+	}
+
+	const removeListener = async (kind: OnMessage, callback: any) => {
+		switch (kind) {
+			case 'accountsChanged':
+				window.interceptor.onAccountsChangedCallBacks.delete(callback)
```

Comment:
Note: I'm still not a fan of using globals and would like to see something like dependency injection via function parameters used instead.